diff --git a/node_modules/react-native-gifted-chat/lib/MessageContainer.js b/node_modules/react-native-gifted-chat/lib/MessageContainer.js
index 6bdf6da..22c5d01 100644
--- a/node_modules/react-native-gifted-chat/lib/MessageContainer.js
+++ b/node_modules/react-native-gifted-chat/lib/MessageContainer.js
@@ -88,6 +88,9 @@ export default class MessageContainer extends React.PureComponent {
             }
         };
         this.handleOnScroll = (event) => {
+            if (this.props.listViewProps.onScrollEvent){
+                this.props.listViewProps.onScrollEvent(event);
+            }
             const { nativeEvent: { contentOffset: { y: contentOffsetY }, contentSize: { height: contentSizeHeight }, layoutMeasurement: { height: layoutMeasurementHeight }, }, } = event;
             const { scrollToBottomOffset } = this.props;
             if (this.props.inverted) {
