diff --git a/node_modules/react-native-screens/ios/bottom-tabs/host/RNSBottomTabsHostEventEmitter.mm b/node_modules/react-native-screens/ios/bottom-tabs/host/RNSBottomTabsHostEventEmitter.mm
index 269895c..8214611 100644
--- a/node_modules/react-native-screens/ios/bottom-tabs/host/RNSBottomTabsHostEventEmitter.mm
+++ b/node_modules/react-native-screens/ios/bottom-tabs/host/RNSBottomTabsHostEventEmitter.mm
@@ -39,7 +39,7 @@ - (BOOL)emitOnNativeFocusChange:(OnNativeFocusChangePayload)payload
   if (_reactEventEmitter != nullptr) {
     _reactEventEmitter->onNativeFocusChange(
         {.tabKey = RCTStringFromNSString(payload.tabKey),
-         .repeatedSelectionHandledBySpecialEffect = payload.repeatedSelectionHandledBySpecialEffect});
+         .repeatedSelectionHandledBySpecialEffect = static_cast<bool>(payload.repeatedSelectionHandledBySpecialEffect)});
     return YES;
   } else {
     RCTLogWarn(@"[RNScreens] Skipped OnNativeFocusChange event emission due to nullish emitter");
